app/src/main/java/com/example/habitsmasher/LoginFragment.java

```diff
@@ -0,0 +1,225 @@
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+/**
+ * This class holds the front-end elements relating to the Login screen
+ * Author: Rudy Patel
+ */
+public class LoginFragment extends Fragment {
+    private static final String TAG = "LoginFragment";
+    private static final String USER_DATA_PREFERENCES_TAG = "USER_DATA";
+    private static final String INCORRECT_EMAIL_PASSWORD_MESSAGE = "Incorrect email/password";
+    private static final String LOGIN_SUCCESSFUL_MESSAGE = "Login successful!";
+    private static final String USERNAME_SHARED_PREF_TAG = "username";
+    private static final String USER_ID_SHARED_PREF_TAG = "userId";
+    private static final String USER_PASSWORD_SHARED_PREF_TAG = "password";
+    private static final String USER_EMAIL_SHARED_PREF_TAG = "email";
+
+    private final LoginFragment _fragment = this;
+    private FirebaseAuth _auth;
+    private ProgressBar _progressBar;
+    private View _bottomNav;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater,
+                             @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.user_login, container, false);
+
+        _auth = FirebaseAuth.getInstance();
+
+        // remove bottom nav bar from login screen
+        _bottomNav = getActivity().findViewById(R.id.nav_view);
+        _bottomNav.setVisibility(GONE);
+
+        View actionBar = getActivity().findViewById(R.id.action_bar);
+        actionBar.setBackgroundColor(getResources().getColor(R.color.habit_list_text));
+
+        // getting UI elements
+        Button loginButton = view.findViewById(R.id.login_button);
+        EditText emailInput = view.findViewById(R.id.login_email);
+        EditText passwordInput = view.findViewById(R.id.login_password);
+        TextView forgotPassword = view.findViewById(R.id.login_forgot_password);
+        _progressBar = view.findViewById(R.id.login_progress_bar);
+
+        setClickListenerForLoginButton(loginButton, emailInput, passwordInput);
+
+        setClickListenerForRegisterButton(view);
+
+        return view;
+    }
+
+    /**
+     * This method sets the click listener for the login button, along with calling the associated
+     * validators and setting/hiding UI related elements
+     * @param loginButton the login button
+     * @param emailInput the email input box
+     * @param passwordInput the password input box
+     */
+    private void setClickListenerForLoginButton(Button loginButton, EditText emailInput, EditText passwordInput) {
+        loginButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
Why initialize the `loginButton` in the main function and pass it as a parameter, when you've initialized the `registerButton` in the `setClickListenerForRegisterButton` function?

Reply:
Something I overlooked when extracting these methods... I can move the register button initialization to the main function instead to match this method. 

Good catch. Fixed!